components/dashboard/src/components/UsageBasedBillingConfig.tsx

```diff
@@ -42,48 +69,42 @@ export default function UsageBasedBillingConfig({ attributionId }: Props) {
     const [currentUsage, setCurrentUsage] = useState<number>(0);
     const [usageLimit, setUsageLimit] = useState<number>(0);
     const [stripePortalUrl, setStripePortalUrl] = useState<string | undefined>();
-    const [pollStripeSubscriptionTimeout, setPollStripeSubscriptionTimeout] = useState<NodeJS.Timeout | undefined>();
-    const [pendingStripeSubscription, setPendingStripeSubscription] = useState<PendingStripeSubscription | undefined>();
     const [errorMessage, setErrorMessage] = useState<string | undefined>();
 
-    const localStorageKey = `pendingStripeSubscriptionFor${attributionId}`;
-    const now = dayjs().utc(true);
+    const now = useMemo(() => dayjs().utc(true), []);
     const [billingCycleFrom, setBillingCycleFrom] = useState<dayjs.Dayjs>(now.startOf("month"));
     const [billingCycleTo, setBillingCycleTo] = useState<dayjs.Dayjs>(now.endOf("month"));
 
-    const refreshSubscriptionDetails = async (attributionId: string) => {
-        setStripeSubscriptionId(undefined);
-        setIsLoadingStripeSubscription(true);
-        try {
-            getGitpodService().server.findStripeSubscriptionId(attributionId).then(setStripeSubscriptionId);
-            const costCenter = await getGitpodService().server.getCostCenter(attributionId);
-            setUsageLimit(costCenter?.spendingLimit || 0);
-            setBillingCycleFrom(dayjs(costCenter?.billingCycleStart || now.startOf("month")).utc(true));
-            setBillingCycleTo(dayjs(costCenter?.nextBillingTime || now.endOf("month")).utc(true));
-        } catch (error) {
-            console.error("Could not get Stripe subscription details.", error);
-            setErrorMessage(`Could not get Stripe subscription details. ${error?.message || String(error)}`);
-        } finally {
-            setIsLoadingStripeSubscription(false);
-        }
-    };
+    const refreshSubscriptionDetails = useCallback(
+        async (attributionId: string) => {
+            setStripeSubscriptionId(undefined);
+            setIsLoadingStripeSubscription(true);
+            try {
+                getGitpodService().server.getStripePortalUrl(attributionId).then(setStripePortalUrl);
+                getGitpodService().server.getUsageBalance(attributionId).then(setCurrentUsage);
+                const costCenter = await getGitpodService().server.getCostCenter(attributionId);
+                setUsageLimit(costCenter?.spendingLimit || 0);
+                setBillingCycleFrom(dayjs(costCenter?.billingCycleStart || now.startOf("month")).utc(true));
+                setBillingCycleTo(dayjs(costCenter?.nextBillingTime || now.endOf("month")).utc(true));
+                const subscriptionIdPromise = await getGitpodService().server.findStripeSubscriptionId(attributionId);
```

Comment:
since we're awaiting, this isn't a promise is it, is it actually a subscription id?